src/runner/browser-job.js

```diff
@@ -10,28 +10,35 @@ const QUARANTINE_THRESHOLD = 3;
 
 // Browser job
 export default class BrowserJob extends EventEmitter {
-    constructor (tests, browserConnection, proxy, screenshots, warningLog, fixtureHookController, opts) {
+    constructor (tests, browserConnections, proxy, screenshots, warningLog, fixtureHookController, opts) {
         super();
 
-        this.started    = false;
-        this.quarantine = null;
+        this.started = false;
 
         this.total                 = 0;
         this.passed                = 0;
         this.opts                  = opts;
         this.proxy                 = proxy;
-        this.browserConnection     = browserConnection;
+        this.browserConnections    = browserConnections;
         this.screenshots           = screenshots;
         this.warningLog            = warningLog;
         this.fixtureHookController = fixtureHookController;
         this.result                = null;
 
-        this.testRunQueue    = tests.map((test, index) => this._createTestRun(test, index + 1, 1));
+        this.testsQueue = tests.map((test, index) => ({
+            test,
```

Comment:
Create class or factory method for this object. It's purpose is absolutely unclear and we should isolate it's construction point.